src/main/java/frc/robot/EagleUtil.java

```diff
@@ -94,7 +156,7 @@ public static ArrayList<Pose2d> calculateBlueReefSetPoints() {
    */
   public static ArrayList<Pose2d> calculateRedReefSetPoints() {
     if (m_redPoses != null) {
-      return m_redPoses;
+      return new ArrayList<Pose2d>(m_redPoses);
```

Comment:
Also do this at the end of this function, so 100% of the calls have a safe copy.
Same on the Blue version